jerry-core/jerryscript.h

```diff
@@ -322,9 +335,19 @@ jerry_value_t jerry_get_object_keys (const jerry_value_t obj_val);
 jerry_value_t jerry_get_prototype (const jerry_value_t obj_val);
 jerry_value_t jerry_set_prototype (const jerry_value_t obj_val, const jerry_value_t proto_obj_val);
 
+__attr_deprecated___
 bool jerry_get_object_native_handle (const jerry_value_t obj_val, uintptr_t *out_handle_p);
+__attr_deprecated___
 void jerry_set_object_native_handle (const jerry_value_t obj_val, uintptr_t handle_p,
                                      jerry_object_free_callback_t freecb_p);
+
+bool jerry_get_object_native_pointer (const jerry_value_t obj_val,
+                                      uintptr_t *out_native_p,
+                                      const jerry_object_native_info_t **out_info_p);
+void jerry_set_object_native_pointer (const jerry_value_t obj_val,
+                                      uintptr_t native_p,
```

Comment:
> in order to backward compatible , we need to define another new function type

Yeah, I think that's OK, no?

> then we need to provide another jerry_dispatch_object_free_callback

Hmm.. internally, the pointer to the callback itself is a `ecma_external_pointer_t` (aka `uintptr_t` and a cast to `jerry_object_free_callback_t` happens in `jerry_dispatch_object_free_callback` itself. I guess as long as `sizeof(uintptr_t) == sizeof(void *)` is true, it will work OK. AFAIK, `uintptr_t` and `void *` are meant to be of equal size. So, I think casting between the callback function pointer types should be fine in that case? I guess it's a bit dangerous if the signature of one would ever change. I'm not aware of a `_Static_assert` that would check function pointer signature "compatibility".

Alternatively, if you provide another `jerry_dispatch_object_free_callback`, is the compiler smart enough to merge the two together?

@zherczeg thoughts on this?